ObjectMemory.c

```diff
@@ -1,4 +1,47 @@
 #include "ObjectMemory.h"
+#include "RealWordMemory.h"
+
+Bool ObjectMemory_new(void) {
```

Comment:
This is the real meat of this PR. `ObjectMemory_new`:
 - makes sure there is heap space
 - puts empty lists of small chunks in each heap segment
 - puts all of the available heap memory in each segment into one object which is on the `BigSize` chunk list
 - makes sure all `Interpreter`-defined objects have a non-zero refcount
 - marks all other object pointers as free and adds them to the free pointer list
 - initialises the `currentSegment` register to 0